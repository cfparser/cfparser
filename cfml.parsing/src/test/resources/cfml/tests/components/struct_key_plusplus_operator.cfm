	foo = { bar: 1 };
	foo.bar++;
