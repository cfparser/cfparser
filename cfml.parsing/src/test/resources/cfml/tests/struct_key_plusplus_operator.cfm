<cfscript>
	foo = { bar: 1 };
	foo.bar++;
</cfscript>
