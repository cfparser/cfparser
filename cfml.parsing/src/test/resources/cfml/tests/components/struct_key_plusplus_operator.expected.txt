/*===TOKENS===*/
IDENTIFIER               <foo>
'='                      <=>
'{'                      <{>
IDENTIFIER               <bar>
':'                      <:>
INTEGER_LITERAL          <1>
'}'                      <}>
';'                      <;>
IDENTIFIER               <foo>
'.'                      <.>
IDENTIFIER               <bar>
'++'                     <++>
';'                      <;>
/*===TREE===*/
(scriptBlock
  (element
    (statement
      (assignmentExpression
        (startExpression
          (compareExpression
            (baseExpression (unaryExpression (memberExpression (identifier foo))))
          )
        )
        =
        (startExpression
          (compareExpression
            (baseExpression
              (unaryExpression
                (primaryExpression
                  (implicitStruct
                    {
                    (implicitStructElements
                      (implicitStructExpression
                        (implicitStructKeyExpression (multipartIdentifier (identifier bar)))
                        :
                        (baseExpression (unaryExpression (primaryExpression (literalExpression 1))))
                      )
                    )
                    }
                  )
                )
              )
            )
          )
        )
      )
      ;
    )
  )
  (element
    (statement
      (startExpression
        (compareExpression
          (baseExpression
            (unaryExpression
              (unaryExpression (memberExpression (identifier foo) . (identifier bar)))
              ++
            )
          )
        )
      )
      ;
    )
  )
  (endOfScriptBlock <EOF>)
)
/*======*/